device/samples/javascript/device_through_proxy.js

```diff
@@ -23,7 +23,10 @@ const deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
 let sendInterval;
 
 function disconnectHandler () {
-  clearInterval(sendInterval);
+  if (!!sendInterval) {
+    clearInterval(sendInterval);
+    sendInterval = null;
+  }
```

Comment:
The double bang isn't needed here. Same feedback applies to the other samples.
```suggestion
  if (sendInterval) {
    clearInterval(sendInterval);
    sendInterval = null;
  }
```